docs/docs/integration/supreme.md

```diff
@@ -69,253 +87,233 @@ As shown in Figure&nbsp;1, the back-end needs to be configured before being able
+    Warden Supreme provides a secure nonce generation service and uses an in-memory challenge cache by default, which is fine for small to medium load but not for larger production deployments.
+    In such scenarios, roll your own, backed by a RDBMS!
 
 ```kotlin
-val ENDPOINT_CHALLENGE = "/api/v1/challenge"
-val ENDPOINT_ATTEST = "/api/v1/attest"
-val PROOF_OID = ObjectIdentifier(Uuid.parse("c893b702-28f6-4c50-8578-d1d7a1580729"))
+--8<-- "Readme-Verifier-min.kt:3"
 ```
 
-#### Back-End Setup
-In addition to the parameters described above, the back-end also needs a source to generate attestation challenges, track them, and match them against incoming attestation requests.
-As Warden Supreme's verifier component aims to integrate with any service, it simply expects a lambda that matches an incoming attestation statement against the expected nonce.
-Session management is out of scope, as it is provided by frameworks such as Ktor or Spring.
-In the end, a verifier instance is created as follows:
+1. Yes, it really is that simple 99% of the time!
 
-```kotlin
-val attestationValidator = AttestationValidator(
-    warden, /* the configured instance as per Back-End Configuration section */
-    attestationProofOID = PROOF_OID
-) {
-    /*
-    Your nonce validation logic here:
-    Usually, you'll want to check
-      * whether the challenge you got is one you issued before
-      * and whether it is still fresh enough
-    and then remove it from whatever challenge-cache you are using.
+??? example "Comprehensive list of Verifier options"
+    ```kotlin
+    --8<-- "Readme-Verifier.kt:18"
+    ```
     
-    Since you receive the challenge in the logic attached to the HTTP endpoint accepting attestation
-    statements, you'll be matching it against the active session there anyway.
-    */
-}
-```
-
-As mentioned, it is also possible to pass the configuration parameters directly.
-
-####  Handling Requests
-
-<h6> <code>ENDPOINT_CHALLENGE</code> is expected to return an <code>AttestationChallenge</code></h6>
-
-The encoding of the challenge is not fixed, but can be serialized using `kotlinx.serialization`
-
-`AttestationChallenge`, contains the following properties:
-1. `nonce: ByteArray`: The actual challenge value; usually a cryptographic nonce, based on true randomness
-2. `validity: Duration`: This is used to communicate the validity duration of a challenge to the client
-3. `timeZone: TimeZone`: Optional information about the TimeZone set on the backend
-4. `postEndpoint: String`: This property conveys the endpoint to post the attestation statement to. This will typically be `<service url>/$ENDPOINT_ATTEST`.
-5. `timeOffset: Duration`: This property is used to inform the client about the maximum tolerated time offset for temporal validations.
-
-A fresh challenge is prepared as follows:
+    1. We want Warden Supreme to conveys the attestation proof inside the CSR using a custom OID.
```

Comment:
```suggestion
    1. We want Warden Supreme to convey the attestation proof inside the CSR using a custom OID.
```
But the code actually uses the default OID!? Also, how would we set a custom OID? By string? a custom data class?